nms/app/components/insights/AsyncMetric.tsx

```diff
@@ -204,34 +211,22 @@ function getUnit(timeRange: TimeRange) {
   return RANGE_VALUES[timeRange].unit;
 }
 
-function getStepUnit(
-  startEnd: [moment.Moment, moment.Moment],
-): [string, TimeUnit] {
+function getStepUnit(startEnd: [Date, Date]): [string, TimeUnit] {
   const [start, end] = startEnd;
-  const d = moment.duration(end.diff(start));
-  const hrs = d.asHours();
-  const days = d.asDays();
-  let r: RangeValue;
-  if (hrs <= 24) {
-    if (hrs <= 3) {
-      r = RANGE_VALUES['3_hours'];
-    } else if (hrs <= 6) {
-      r = RANGE_VALUES['6_hours'];
-    } else if (hrs <= 12) {
-      r = RANGE_VALUES['12_hours'];
-    } else {
-      r = RANGE_VALUES['24_hours'];
-    }
-  } else {
-    if (days <= 7) {
-      r = RANGE_VALUES['7_days'];
-    } else if (days <= 14) {
-      r = RANGE_VALUES['14_days'];
-    } else {
-      r = RANGE_VALUES['30_days'];
+  const durationInHours = differenceInHours(end, start);
+  const durationInDays = differenceInDays(end, start);
+
+  let range: RangeValue = RANGE_VALUES['3_hours'];
+  let range_item: keyof typeof RANGE_VALUES;
+  for (range_item in RANGE_VALUES) {
```

Comment:
Sorry to go on about this specific change, but I think the order of `for...in` loops is implementation-dependent and not guaranteed, so this might be dangerous even if it works in a test.

Reply:
You might be right. So be it than. (But I would have really loved to have a proper implementation without this manual assignment and duplication.)